### PR TITLE
WIP:When LoadStore a uint32, return value should be uint32

### LINKED ISSUE #215: [QEMU] === cctest/test-assembler-riscv64/RISCV_UTEST_swlwu === failed
**Describe the bug**
failed

**To Reproduce**
1. Build with options cross-build
2. Run test/program '...'
3. See error

**Expected behavior**

**Errors/Logs**
```
=== cctest/test-assembler-riscv64/RISCV_UTEST_swlwu ===
--- stderr ---
#
# Fatal error in ../../test/cctest/test-assembler-riscv64.cc, line 134
# Check failed: converted_res == expected_res (2238380180 vs. 2238380180).
#
#
#
#FailureMessage Object: 0x3fffc3d238
==== C stack trace ===============================

    /root/riscv64.native.debug/cctest(+0x3461a68) [0x2ac58b3a68]
    /root/riscv64.native.debug/cctest(+0x33b8412) [0x2ac580a412]
    /root/riscv64.native.debug/cctest(+0x31a1d82) [0x2ac55f3d82]
    /root/riscv64.native.debug/cctest(+0xfcb7c8) [0x2ac341d7c8]
    /root/riscv64.native.debug/cctest(+0xfd6b10) [0x2ac3428b10]
    /root/riscv64.native.debug/cctest(+0x89781e) [0x2ac2ce981e]
    /root/riscv64.native.debug/cctest(+0x81d74c) [0x2ac2c6f74c]
    /lib64/lp64d/libc.so.6(__libc_start_main+0xa0) [0x3fc31732a2]
    /root/riscv64.native.debug/cctest(_start+0x2c) [0x2ac2ce7228]
Command: /root/riscv64.native.debug/cctest test-assembler-riscv64/RISCV_UTEST_swlwu --random-seed=-617479117 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```

### PR DESCRIPTION
**Pull request by luyahan**
fix #215

### CODE REVIEW

**Comment by luyahan** [comment]
fix #215

File: test/cctest/test-assembler-riscv64.cc [hunk 1]
```diff
@@ -292,6 +292,34 @@ static void GenAndRunTestForLoadStore(T value, Func test_generator) {
```
Comment (lazyparser) [inline test/cctest/test-assembler-riscv64.cc:292]: hello from PLCT ci bot

**Review by penguinwu — CHANGES_REQUESTED** [review]
LGTM.

But there are a few other `GenAndRunTest` functions that also have a similar problem of potentially returning int32/64 for uint32/64 input types. Search for string `int32_t, int64_t`. Please separate the floating-point version from the integer version and fix those too.

**Comment by penguinwu** [comment]
@luyahan Let's put this PR on hold (no need to address my change request). I am finishing up a pretty extensive refactoring PR for `test-assembler/macro-assembler-riscv64.cc` that touches the same `GenAndRunTest` functions that you modified here. I've incorporated the fixes (without duplicating GenAndRunTest functions for floating-point) for all similar cases in my PR. Let me merge my PR first. If the problem still persists, you can apply your fix on top of mine.

**Comment by luyahan** [comment]
> @luyahan Let's put this PR on hold (no need to address my change request). I am finishing up a pretty extensive refactoring PR for `test-assembler/macro-assembler-riscv64.cc` that touches the same `GenAndRunTest` functions that you modified here. I've incorporated the fixes (without duplicating GenAndRunTest functions for floating-point) for all similar cases in my PR. Let me merge my PR first. If the problem still persists, you can apply your fix on top of mine.

That's the best !

**Comment by penguinwu** [comment]
@luyahan My PR  is merged. Please try the latest from riscv64 to see if it fixes the problem you reported. If so, please close the corresponding issue.

**Pull request closed without merging**